packages/perspective-viewer-d3fc/src/js/d3fcChart.js

```diff
@@ -50,19 +50,29 @@ export default class D3FCChart {
 
         let labels = interpretLabels(config);
         let isSplitBy = labels.splitLabel.length !== 0;
+        const isMultiColumn = interpretIsMultiColumn(config);
 
         let groupBys = interpretGroupBys(config.xAxis.categories);
         let series = config.series;
 
-        let [dataset, stackedBarData, color] = interpretDataset(isSplitBy, series, groupBys, hiddenElements);
+        const [keys, color] = interpretKeysAndColor(config);
+        let [dataset, stackedBarData] = interpretDataset(isSplitBy, series, groupBys, hiddenElements);
 
-        let legend = configureLegend(isSplitBy, color, hiddenElements, update);
+        let legend = configureLegend(config.legend.enabled, color, hiddenElements, update);
         let barSeries = configureBarSeries(isSplitBy, orientation, dataset);
         let gridlines = configureGrid(horizontal);
         let [xScale, yScale] = configureScale(isSplitBy, horizontal, dataset, stackedBarData);
-
         // groups of svgs we need to render
         let multi = configureMultiSvg(isSplitBy, gridlines, barSeries, dataset, color);
+        let groupedBarData;
+
+        if (isMultiColumn) {
+            [dataset, groupedBarData] = interpretMultiColumnDataset(config, hiddenElements);
+            barSeries = configureMultiColumnBarSeries(orientation, color, keys);
+            [xScale, yScale] = configureScaleMultiColumn(horizontal, dataset, groupedBarData);
```

Comment:
@DevAndyLee , you are correct, I shouldn't be setting the values and then overriding them. The reason I did this was because I wanted to leave the original code until I have worked out how to combine the two types of chart. When I am finished there shouldn't be two assignments. This is intended as a temporary measure.